### PR TITLE
Adding support for optional proxy server on REST invoke

### LINKED ISSUE #166: Add optional proxy settings for rest activity
The REST activity should have optional proxy settings.

### PR DESCRIPTION
**Pull request by mellistibco**
support for rest activity proxy server usage. Resolves #166

### CODE REVIEW

File: activity/rest/activity.go [hunk 1]
```diff
@@ -119,7 +120,21 @@ func (a *RESTActivity) Eval(context activity.Context) (done bool, err error) {
 		req.Header.Set("Content-Type", contentType)
 	}
 
-	client := &http.Client{}
+	// Set the proxy server to use, if supplied
+	proxy := context.GetInput(ivProxy)
+	var client *http.Client
+	if proxy != nil && proxy.(string) != "" {
```
Comment (torresashjian) [inline activity/rest/activity.go]: You should use
v, ok := proxy.(string)

More here: https://golang.org/ref/spec#Type_assertions

File: activity/rest/activity.go [hunk 2]
```diff
@@ -119,7 +120,21 @@ func (a *RESTActivity) Eval(context activity.Context) (done bool, err error) {
...
+		proxyURL, urlErr := url.Parse(proxy.(string))
+		if urlErr != nil {
+			log.Debug("Error parsing proxy url, skipping proxy:", urlErr)
+			client = &http.Client{}
```
Comment (torresashjian) [inline activity/rest/activity.go]: Are we allowing to continue with at bad proxy? can't this bring undesired behavior?

**Comment by mellistibco** [comment]
thanks @torresashjian updated to reflect your suggestions

File: activity/rest/activity.go [hunk 3]
```diff
@@ -119,7 +120,22 @@ func (a *RESTActivity) Eval(context activity.Context) (done bool, err error) {
 		req.Header.Set("Content-Type", contentType)
 	}
 
-	client := &http.Client{}
+	// Set the proxy server to use, if supplied
+	proxy := context.GetInput(ivProxy)
+	var client *http.Client
+	var proxyValue, ok = proxy.(string)
```
Comment (torresashjian) [inline activity/rest/activity.go]: the ok pattern will take care of the nil, this should suffice, also for empty strings len() is used:
 proxyValue, ok := proxy.(string)
if ok && len(proxyValue) > 0{
...
} else {
...
}

**Pull request merged**